test/hotspot/jtreg/runtime/posixSig/TestPosixSig.java

```diff
@@ -44,8 +44,16 @@ public static void main(String[] args) throws Throwable {
         if (args.length == 0) {
 
             // Create a new java process for the TestPsig Java/JNI test.
+            // We run the VM in interpreted mode, because the JIT might mark
+            // a Java method as not-entrant, which means turning the first opcode
+            // of the compiled method to NULL. Calling such a method after establishing
```

Comment:
This comment is PPC64 specific. Please make it more general. That the instruction "0" generates SIGILL is specified by the PPC64 ISA and may be wrong for other platforms. Better would be "which means turning the first instruction into an illegal one".

Reply:
Also would be good to mention that the problem is one of time: "If a SIGILL arrives after we redirected the signal handler but before the JNI checker noted the signal handler modification, the JVM may crash or hang; since SIGILLs may be generated by compiled code, we run interpreted".